plugins/modules/ec2_vpc_nat_gateway.py

```diff
@@ -448,193 +394,191 @@ def gateway_in_subnet_exists(client, subnet_id, allocation_id=None,
     Returns:
         Tuple (list, bool)
     """
+
     allocation_id_exists = False
     gateways = []
     states = ['available', 'pending']
 
-    gws_retrieved, err_msg, gws = (
-        get_nat_gateways(
-            client, subnet_id, states=states, check_mode=check_mode
-        )
-    )
+    gws_retrieved = (get_nat_gateways(client, module, subnet_id, states=states))
 
-    if not gws_retrieved:
-        return gateways, allocation_id_exists
-    for gw in gws:
-        for address in gw['nat_gateway_addresses']:
-            if allocation_id:
-                if address.get('allocation_id') == allocation_id:
-                    allocation_id_exists = True
+    if gws_retrieved:
+        for gw in gws_retrieved:
+            for address in gw['nat_gateway_addresses']:
+                if allocation_id:
+                    if address.get('allocation_id') == allocation_id:
+                        allocation_id_exists = True
+                        gateways.append(gw)
+                else:
                     gateways.append(gw)
-            else:
-                gateways.append(gw)
 
     return gateways, allocation_id_exists
 
 
-def get_eip_allocation_id_by_address(client, eip_address, check_mode=False):
+def get_eip_allocation_id_by_address(client, module, eip_address):
     """Release an EIP from your EIP Pool
     Args:
         client (botocore.client.EC2): Boto3 client
+        module: AnsibleAWSModule class instance
         eip_address (str): The Elastic IP Address of the EIP.
 
-    Kwargs:
-        check_mode (bool): if set to true, do not run anything and
-            falsify the results.
-
     Basic Usage:
         >>> client = boto3.client('ec2')
+        >>> module = AnsibleAWSModule(...)
         >>> eip_address = '52.87.29.36'
-        >>> get_eip_allocation_id_by_address(client, eip_address)
+        >>> get_eip_allocation_id_by_address(client, module, eip_address)
         'eipalloc-36014da3'
 
     Returns:
         Tuple (str, str)
     """
+
     params = {
         'PublicIps': [eip_address],
     }
     allocation_id = None
-    err_msg = ""
+    msg = ''
 
     try:
         allocations = client.describe_addresses(aws_retry=True, **params)['Addresses']
+
         if len(allocations) == 1:
             allocation = allocations[0]
         else:
             allocation = None
 
         if allocation:
             if allocation.get('Domain') != 'vpc':
-                err_msg = (
+                msg = (
                     "EIP {0} is a non-VPC EIP, please allocate a VPC scoped EIP"
                     .format(eip_address)
                 )
             else:
                 allocation_id = allocation.get('AllocationId')
         else:
-            err_msg = (
+            msg = (
                 "EIP {0} does not exist".format(eip_address)
             )
+    except is_boto3_error_code('InvalidAddress.Malformed') as e:
+        allocation_id = None
```

Comment:
If the EIP the user provides is malformed we should probably return that information to the user so that they can troubleshoot their playbook, we're returning an empty `msg` now.  For a malformed EIP we should probably fail here rather than returning back to pre_create since we can't proceed with the information provided in the playbook.

Reply:
I added fail_json for this case. Let me know if this works for you. Thank you @jillr.